server/server.go

```diff
@@ -291,10 +291,16 @@ func NewServer(ctx context.Context, opts ArgoCDServerOpts) *ArgoCDServer {
 	apiFactory := api.NewFactory(settings_notif.GetFactorySettings(argocdService, "argocd-notifications-secret", "argocd-notifications-cm"), opts.Namespace, secretInformer, configMapInformer)
 
 	dbInstance := db.NewDB(opts.Namespace, settingsMgr, opts.KubeClientset)
+	logger := log.NewEntry(log.StandardLogger())
+
+	sg := extension.NewDefaultSettingsGetter(settingsMgr)
+	ag := extension.NewDefaultApplicationGetter(appLister)
+	pg := extension.NewDefaultProjectGetter(projLister, dbInstance)
+	em := extension.NewManager(logger, sg, ag, pg, enf)
```

Comment:
Is there any performance downside to setting these up? Currently they're only set up if proxy extensions are enabled. With this change, they'd be initialized regardless.

Reply:
This code is pretty much just doing dependency injection. There won't be any performance impact while executing this code.